### PR TITLE
various updates to address comments, etc.

### PR DESCRIPTION
**Pull request by scott-vsi**

### CODE REVIEW

File: linux/just_files/just_functions.bsh [hunk 1]
```diff
@@ -299,10 +297,9 @@ function source_environment_files()
 
   export JUST_SETTINGS
 
-  local JUST_LOCAL_SETTINGS
-  local JUST_LOCAL_SETTINGS_POST
-  # Sets JUST_LOCAL_SETTINGS and JUST_LOCAL_SETTINGS_POST
-  try_to_create_local_settings_file "${1}" || : # <== Fix for #84
+  local JUST_LOCAL_SETTINGS="${JUST_LOCAL_SETTINGS-}"
+  local JUST_LOCAL_SETTINGS_POST="${JUST_LOCAL_SETTINGS_POST-}"
```
Comment (andyneff) [inline linux/just_files/just_functions.bsh:301]: What happened to the default values here?

File: linux/parser.bsh [hunk 2]
```diff
@@ -18,6 +18,28 @@ fi
...
+#
+# .. warning::
+#
+#   Parsing URLs 100% correctly is HARD. This is not that solution.
+#
+# .. note::
+#    From https://stackoverflow.com/a/45977232
+#**
+
+JUST_URI_REGEX='^(([^:/?#]+):)?(//((([^:/?#]+)@)?([^:/?#]+)(:([0-9]+))?))?((/|$)([^?#]*))(\?([^#]*))?(#(.*))?$'
```
Comment (andyneff) [inline linux/parser.bsh:36]: This isn't a JUST thing, so you could rename it...

URI_REGEX
VSI_URI_REGEX
VSI_COMMON_URI_REGEX
PARSER_URI_REGEX
etc...

File: linux/parser.bsh [hunk 3]
```diff
@@ -41,41 +63,23 @@ fi
 #   ``just_url_parsed_rpath`` - e.g., vsi_common/just/quickstart_user.html
 #   ``just_url_parsed_query`` - e.g., highlight=J.U.S.T
 #   ``just_url_parsed_fragment`` - e.g., what-is-j-u-s-t
```
Comment (andyneff) [inline linux/parser.bsh:65]: just_->something else, or remove

File: tests/int/test-git_mirror.bsh [hunk 4]
```diff
@@ -384,3 +333,46 @@ begin_test "Part 7 - Update a repo and the prepped mirror"
   git -C "${MIRROR_DIR}"/main show-ref --verify --quiet refs/heads/new_branch
 )
 end_test
+
+function begin_archive_test()
+{
+  tar_version_info # Sets tar_version and tar_vendor
+  if [ "${tar_vendor}" == "GNU" ] || \
+    ([ "${tar_vendor}" == "BSD" ] && meet_requirements "${tar_version}" ">=3"); then
+    # ugh
```
Comment (andyneff) [inline tests/int/test-git_mirror.bsh:342]: Remove

File: linux/just_git_airgap_repo.bsh [hunk 5]
```diff
@@ -1062,96 +1204,56 @@ function relocate_git_defaultify()
...
-      # Also have to handle the --no- variant manually
-      while (( $# )); do
-        # Convert arguments of the form -f=val to -f val and --flag=val to --flag val
-        # https://medium.com/@binaryphile/nice-one-elegant-in-its-simplicity-eee6b433bc90
-        if [[ $1 == -?=* ]] || [[ $1 == --*=* ]]; then
-          set -- "${1%%=*}" "${1#*=}" "${@:2}"
+
+      local parsed_args
+      local usage_help=0
+      # TODO Support --remote-url (but disallow also setting --check-remote)
```
Comment (andyneff) [inline linux/just_git_airgap_repo.bsh:1217]: Use an if statement after parse_args to see if both args were specified.

File: linux/just_git_airgap_repo.bsh [hunk 6]
```diff
@@ -1062,96 +1204,56 @@ function relocate_git_defaultify()
...
+          -c check_remote --check-remote check_remote --no-check-remote check_remote! \
+          -r remote_name: --remote-name remote_name: -p prep_dir: --prep-dir prep_dir: \
+          -o output_dir: --output-dir output_dir: -- ${@+"${@}"}
+      shift "${parsed_args}"
+      if [ $# -gt 0 ]; then
+        if [[ "$1" = "-"* ]]; then
+          echo "Unknown flag: $1"
+        else
+          echo "Positional arguments are not allowed: $1"
         fi
```
Comment (andyneff) [inline linux/just_git_airgap_repo.bsh:1229]: You are checking to see if there are any unparsed args here.

Just keep in mind that "chained just targets" would have resulted in unparsed flags here, so you are indirectly saying you can't have chained a target with this target. That is allowed, I just want you to be aware what this is really doing.

File: linux/just_git_airgap_repo.bsh [hunk 7]
```diff
@@ -1062,96 +1204,56 @@ function relocate_git_defaultify()
...
+        echo "   -b, --branch         The branch on which to base the mirror; e.g., master"
+        echo "   -c, check-remote     Ensure that changes to a submodule that are tracked"
+        echo "   --no-check-remote    by its parent have been pushed to the submodule's URL"
+        echo "   -h, --help           Print this help"
+        echo "   -p, --prep-dir       The path to a previously mirrored repository"
+        echo "   -o, --output-dir     The output directory in which to mirror the repositories"
+        echo "   -r, --remote-name    The git remote (with associated URL) on which to base the"
+        echo "                        mirror; e.g., origin"
+        echo
+        exit 0
```
Comment (andyneff) [inline linux/just_git_airgap_repo.bsh:1247]: `exit 0` -> `return 0`

File: linux/just_git_airgap_repo.bsh (hunk 5 above)
Comment (andyneff) [reply linux/just_git_airgap_repo.bsh:1217]: You already do this for --prep_dir and --output_dir

File: linux/just_files/just_functions.bsh (hunk 1 above)
Comment (scott-vsi) [reply linux/just_files/just_functions.bsh:301]: They are set in create_local_settings_file

File: linux/just_git_airgap_repo.bsh [hunk 8]
```diff
@@ -516,7 +670,13 @@ function create_repo_map()
 
   local OLD_IFS="${IFS}"
   IFS=$'\n'
-  # The (relative) path from the top-level repository to each submodule (recursively)
+  # The (relative, i.e., display) path from the project repository to each
+  # submodule (recursively)
+  # NOTE displaypath only provides the expected path if this command is run
```
[the remark this answers is not in the document]
Comment (scott-vsi) [reply linux/just_git_airgap_repo.bsh:675]: `prefix` was deprecated for `displaypath` at some point. I'm actually not sure when, but git 1.8 uses prefix. In git 1.8, `git submodule` must be run from the top-level directory. So `prefix` doesn't have the same flexibility as `displaypath`.

File: linux/just_git_airgap_repo.bsh [hunk 9]
```diff
@@ -604,24 +764,15 @@ function orphan_commit_repo_map
...
-    echo "not empty. This script will delete it and continue."
-    ask_question "Is that ok?" answer_delete n
-    [ "$answer_delete" == "0" ] && return 1
-    rm -rf __just_git_clone
-  fi
-  ${GIT} clone --no-checkout . __just_git_clone
-  pushd __just_git_clone &>/dev/null
+  # Clone into an automatically deleting temporary directory
+  make_temp_path temp_repo -d just.XXXXXXXXXX
+  ${GIT} clone --no-checkout . "${temp_repo}"
```
[the remark this answers is not in the document]
Comment (scott-vsi) [reply linux/just_git_airgap_repo.bsh:769]: Hmm, that's a good idea.

File: tests/int/test-just_git_airgap_repo.bsh [hunk 10]
```diff
@@ -90,23 +104,22 @@ begin_test "Part 1 - Setup test repo"
...
-command -v "${GIT-git}" &> /dev/null || skip_next_test
 begin_test "Part 2 - Initial mirror"
 (
   setup_test
   setup_variables
 
   pushd "${BUILD_REPO}" &> /dev/null
     . "${BUILD_REPO}/vsi_common/linux/just_git_airgap_repo.bsh"
-    # FIXME omitting JUST_VERSION did not cause this test to fail, even though
+    # FIXME: omitting JUST_VERSION did not cause this test to fail, even though
```
[the remark this answers is not in the document]
Comment (scott-vsi) [reply tests/int/test-just_git_airgap_repo.bsh:114]: See #282

**Comment by drewgilliam** [comment]
FYI - appveyor is not actually running the git_mirror test

`setup()` doesn't have unzip, so it silently fails.
```
int/test-git_mirror.bsh summary: 0 tests, 0 failures, 0 unexpected successes, 0 expected failures, 0 required failures, 0 skipped
```

Please add the following to `appveyor.yml` under `install:` to actually run `test-git_modules.bsh`
```
install:
``
  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S unzip"
```

**Pull request merged**